library/src/main/java/org/apache/apex/malhar/lib/state/managed/MovingBoundaryTimeBucketAssigner.java

```diff
@@ -125,9 +125,11 @@ public long getTimeBucket(long time)
       long move = (diffInBuckets + 1) * bucketSpanMillis;
       start += move;
       end += move;
-      lowestPurgeableTimeBucket += diffInBuckets;
       // trigger purge when lower bound changes
-      triggerPurge = (diffInBuckets > 0);
+      triggerPurge = (diffInBuckets >= 0);
```

Comment:
If using `move` seems to be more intuitive, then we can also use `move > 0` to trigger the purge